ipaserver/install/adtrustinstance.py

```diff
@@ -530,27 +530,18 @@ def __setup_group_membership(self):
             api.Backend.ldap2, self.smb_dn, "member",
             [self.cifs_agent, self.host_princ])
 
-    def __setup_principal(self):
-        try:
-            api.Command.service_add(unicode(self.principal))
-        except errors.DuplicateEntry:
-            # CIFS principal already exists, it is not the first time
-            # adtrustinstance is managed
-            # That's fine, we we'll re-extract the key again.
-            pass
-        except Exception as e:
-            self.print_msg("Cannot add CIFS service: %s" % e)
-
+    def _clean_previous_keytab(self):
+        """
+        Purge old CIFS keys from /etc/krb5.keytab and clean up samba ccache
```

Comment:
Samba does not use /etc/krb5.keytab. Instead, Samba has own keytab in /etc/samba/samba.keytab.